app/src/main/java/com/example/habitsmasher/LoginFragment.java

```diff
@@ -74,15 +87,150 @@ public View onCreateView(@NonNull LayoutInflater inflater,
+                Log.d(TAG, "firebaseAuthWithGoogle:" + account.getId());
+                firebaseAuthWithGoogle(account.getIdToken());
+            } catch (ApiException e) {
+                // Google Sign In failed, update UI appropriately
+                Log.w(TAG, "Google sign in failed", e);
+                showMessage("Sign in failed", Toast.LENGTH_SHORT);
+            }
+        }
+    }
+
+    /**
+     * This helper method uses the GoogleAuthProvider with the given id credential to sign in
+     * using Firebase
+     * @param idToken the id token of the associated credential
+     */
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
+        _auth.signInWithCredential(credential)
+             .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                 @Override
+                 public void onComplete(@NonNull Task<AuthResult> task) {
+                     if (task.isSuccessful()) {
+                         // if log in successful, redirect to home page
+                         showMessage(LOGIN_SUCCESSFUL_MESSAGE, Toast.LENGTH_SHORT);
+
+                         DocumentReference userRef = getCurrentUserDocumentReference();
+                         userRef.get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
+                             @Override
+                             public void onSuccess(DocumentSnapshot documentSnapshot) {
+                                 GoogleSignInAccount user = GoogleSignIn.getLastSignedInAccount(
+                                         getContext());
+
+                                 User googleUser = new User(user.getId(),
+                                                            user.getDisplayName()
+                                                                .replaceAll("\\s+",""),
+                                                            user.getEmail(),
+                                                            GOOGLE_USER_PWD);
+
+                                 // once the user is successfully retrieved, add them to the db
+                                 addUserToDatabaseIfUserDoesNotExist(googleUser);
+
+                                 // save their information to be used globally throughout the app
+                                 saveUserInformation(googleUser);
+
+                                 // transition to the home screen
+                                 navigateToFragmentWithAction(R.id.action_navigation_login_to_HomeFragment);
+                             }
+                         });
+                     } else {
+                         showMessage(INCORRECT_EMAIL_PASSWORD_MESSAGE, Toast.LENGTH_LONG);
+                     }
+                     _progressBar.setVisibility(View.GONE);
+                     _bottomNav.setVisibility(View.VISIBLE);
+                 }
+             });
+    }
```

Comment:
Basically, **all** of the methods above are copy-pasted from the official android documentation (https://firebase.google.com/docs/auth/android/google-signin) on how to integrate with Google sign-in.